### PR TITLE
[#1769] feat(partition): Support `listPartitions` for Hive catalog

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Support `listPartitions` for Hive catalog

### Why are the changes needed?

Fix: #1769 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?

tests added

### CODE REVIEW

**Comment by jerryshao** [comment]
@mchades can you please fix the conflicts.

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java [hunk 1]
```diff
@@ -50,15 +53,21 @@ public Response listPartitionNames(
       @PathParam("metalake") String metalake,
       @PathParam("catalog") String catalog,
       @PathParam("schema") String schema,
-      @PathParam("table") String table) {
+      @PathParam("table") String table,
+      @QueryParam("verbose") @DefaultValue("false") boolean verbose) {
```
Comment (jerryshao) [inline server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java:57]: I would suggest to change the quey parameter to "details".

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java [hunk 2]
```diff
@@ -50,15 +53,21 @@ public Response listPartitionNames(
...
             Table loadTable = dispatcher.loadTable(tableIdent);
-            String[] partitionNames = loadTable.supportPartitions().listPartitionNames();
-            return Utils.ok(new PartitionNameListResponse(partitionNames));
+            if (verbose) {
+              Partition[] partitions = loadTable.supportPartitions().listPartitions();
+              return Utils.ok(new PartitionListResponse(toDTOs(partitions)));
+            } else {
+              String[] partitionNames = loadTable.supportPartitions().listPartitionNames();
+              return Utils.ok(new PartitionNameListResponse((partitionNames)));
+            }
```
Comment (jerryshao) [inline server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java:70]: Also I think you miss the UTs here.

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java (hunk 1 above)
Comment (mchades) [reply server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java]: ok

File: server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java (hunk 2 above)
Comment (mchades) [reply server/src/main/java/com/datastrato/gravitino/server/web/rest/PartitionOperations.java:70]: fixed

File: server/src/test/java/com/datastrato/gravitino/server/web/rest/TestPartitionOperations.java [hunk 3]
```diff
@@ -211,6 +211,46 @@ public void testListPartitionNames() {
     Assertions.assertTrue(errorResp2.getMessage().contains("test exception"));
   }
 
+  @Test
+  public void testListPartitions() {
```
Comment (jerryshao) [inline server/src/test/java/com/datastrato/gravitino/server/web/rest/TestPartitionOperations.java:215]: Can you please add a test case returning empty partitions?

Comment (mchades) [reply server/src/test/java/com/datastrato/gravitino/server/web/rest/TestPartitionOperations.java:215]: added to IT

**Review by jerryshao — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]
